core/types/calls.cc

```diff
@@ -2572,7 +2572,7 @@ class Magic_callWithBlock : public IntrinsicMethod {
             auto what = core::errors::Infer::errorClassForUntyped(gs, args.locs.file, receiver->type);
             if (auto e = gs.beginError(args.argLoc(0), what)) {
                 e.setHeader("Call to method `{}` on `{}`", fn.show(gs), "T.untyped");
-                TypeErrorDiagnostics::explainUntyped(gs, e, what, args.fullType, args.originForUninitialized);
+                TypeErrorDiagnostics::explainUntyped(gs, e, what, *receiver, args.originForUninitialized);
```

Comment:
This was a bug revealed by an ENFORCE that triggered after making this change.

We are now calling `untypedBlame` on the thing passed to `explainUntyped`, and `TypePtr::untypedBlame` asserts that the thing it's called on `hasUntyped()` as a sanity check.

This revealed the bug at hand: we were reporting this error if the receiver is untyped, but then explaining it as if it blamed to the `args.fullType`. But in this intrinsic, the `args.fullType` is always `T.class_of(<Magic>)`.

This didn't _really_ cause problems. It would have just made for a confusing error message if it happened, but the `explainUntyped` helper doesn't do anything like attribute where the untyped value came from or anything like that.